processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -188,6 +188,22 @@ public BeanMappingMethod build() {
                             Message.BEANMAPPING_NOT_ASSIGNABLE, resultType, method.getResultType()
                         );
                     }
+                    else if ( resultType.isAbstract() ) {
+                        ctx.getMessager().printMessage(
+                            method.getExecutable(),
+                            beanMappingPrism.mirror,
+                            Message.BEANMAPPING_ABSTRACT,
+                            resultType,
+                            method.getResultType()
+                        );
+                    }
+                }
+                else if ( !method.isUpdateMethod() && method.getReturnType().isAbstract() ) {
```

Comment:
That doesn't feel right. Here, we only check that the manually provided type is not an abstract class, but it could be an interface (which it mustn't be either).
In the linked code, we check that if the return type is an interface, the result type must be manually defined, although it would also have to be defined if the return type is an abstract class. But both would only be true if there was no factory method found, so I'm not so sure if that other place is really correct.

IMHO we should report errors for the following cases:
* The return type is an interface or abstract class AND 
* no factory method can be found AND
* no result type is specified OR the specified result type is an interface or an abstract class

The "interface or abstract class" could also be amended by " a class without a default constructor" (where we currently generate non-compilable code, btw).